src/main/java/org/broadinstitute/hellbender/tools/exome/NormalizeBySampleDepth.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.commons.math3.linear.RealMatrix;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.CopyNumberProgramGroup;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Calculates basic statistics for coverage per targets and per sample.
+ * <p>
+ *   This tool takes a read counts file and outputs a read counts file in which coverage at each target
+ *   and sample is divided by the average coverage of that sample.
+ * </p>
+ * <p>
+ *   The input and output format for the coverage file is described in {@link ReadCountCollectionUtils}.
+ * </p>
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+@CommandLineProgramProperties(
+        summary = "Normalizes read counts by samples' average coverage.",
+        oneLineSummary = "Normalizes read counts by samples' average coverage.",
+        programGroup = CopyNumberProgramGroup.class
+)
+public class NormalizeBySampleDepth extends CommandLineProgram {
+    public static final String WEIGHTED_AVERAGE_LONG_NAME = "weightedAverage";
+    public static final String WEIGHTED_AVERAGE_SHORT_NAME = "w";
+
+    @Argument(
+            doc = "Input coverage file",
+            fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File inputFile;
+
+    @Argument(
+            doc = "Output coverage file",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File outputFile;
+
+    @Argument(
+            doc = "Weighted average with weights proportional to target sizes",
+            fullName = WEIGHTED_AVERAGE_LONG_NAME,
+            shortName = WEIGHTED_AVERAGE_SHORT_NAME,
+            optional = true
+    )
+    protected boolean weightByTargetSize = false;
+
+    @Override
+    public Object doWork() {
+        final ReadCountCollection inputCounts;
+        try {
+            inputCounts = ReadCountCollectionUtils.parse(inputFile);
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile(inputFile, "cannot reach or read the input target coverage file");
+        }
+
+        final RealMatrix counts = inputCounts.counts();
+
+        //kludgy way to deep copy input targets list
+        final List<Target> targets = inputCounts.targets().stream()
```

Comment:
``new ArrayList<>(...)`` makes a copy.  This would be reasonable if you had something that might extend Target, but you want to enforce the copy being a Target.